airflow/providers/databricks/hooks/databricks.py

```diff
@@ -408,3 +410,18 @@ def get_repo_by_path(self, path: str) -> Optional[str]:
                 raise e
 
         return None
+
+    def test_connection(self):
+        """Test the Databricks connectivity from UI"""
+        status, message = False, ''
+        hook = DatabricksHook(databricks_conn_id=self.databricks_conn_id)
+        try:
+            result = hook._do_api_call(endpoint_info=LIST_ZONES_ENDPOINT).get('zones', [])
+            if result:
```

Comment:
Is it possible this can return a false value (no zones?) but not error? If so this would lead to confusing errors to the user I think?

If it's not possible, then don't have the condition.

Reply:
Below are the api responses , it will either return zones(because every valid databricks workspace is associated with a zone) or an exception. 

For a working connection
(base) phanikv@Phanis-MBP ~ % curl -X GET --header "Authorization: Bearer $DATABRICKS_TOKEN" \
https://dbc-xxxxxxxxxxx.cloud.databricks.com/api/2.0/clusters/list-zones
{"zones":["us-east-2b","us-east-2c","us-east-2a"],"default_zone":"us-east-2b"}

Let's say I enter a wrong host name
(base) phanikv@Phanis-MBP ~ % curl -X GET --header "Authorization: Bearer $DATABRICKS_TOKEN" \
https://dbc-xxxxxxxxxxx.cloud.databricks.com/api/2.0/clusters/list-zones 
curl: (6) Could not resolve host: dbc-xxxxxxxxxxxcloud.databricks.com

Reply:
I will make below changes 
- remove the `if result` check 
- not return empty list when it tries to get zones, as it is un-necessary.

Reply:
@ashb I've addressed this in [280ac31](https://github.com/apache/airflow/pull/24617/commits/280ac3144900bfc35865744d6c7ab960e3756eca)